### DISCUSSION TITLE
new-adventures-under-medium-trust/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#32] [Blog comments]
# new-adventures-under-medium-trust/

https://blogs.taiga.nl/martijn/posts/new-adventures-under-medium-trust/

**Comment by disqus-comments-migrator** [comment]
<em>Juan Woang commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 25 2009, 02:14</em>

---

<p>Thanks martijn a lot. I love CUY much, and last time I use the customer's server, so that does not know version 1.6.x work on full trust.</p>

**Comment by disqus-comments-migrator** [comment]
<em>dotnet specialisten commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 28 2009, 02:12</em>

---

<p>Martijn,</p><p>that is good news! thanx! <br>i didn't care much about medium trust because i run my own server. i know that some friends (cuyahoga fans ;-) and shared hosters ;-( ) will be very happy with this news ;-)</p><p>When will v2.0 be released? any planning?</p>

**Comment by disqus-comments-migrator** [comment]
<em>gedw99 commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 01 2009, 03:46</em>

---

<p>great stuff Martin.</p><p>Looking forward to the 2.0 version. <br>I have submitted a patch for the File System Service, let me know if you need any help in integrating it.</p>

**Comment by disqus-comments-migrator** [comment]
<em>Krzysztof Koźmic commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 10 2009, 06:52</em>

---

<p>Martijn,</p><p>I applied your patch a week ago. Let me know if you find any other issues.</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 10 2009, 07:40</em>

---

<p>@Krzysztof, I'm running the latest Castle trunk now and it works fine.</p><p>@Others: the SVN trunk of last week unfortunately contained a wrong Castle build. Fixed it.</p>

**Comment by disqus-comments-migrator** [comment]
<em>samk commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 15 2009, 11:50</em>

---

<p>Fantastic job Martijn!</p><p>Any chance you [or someone else] can publish a patched NHibernate+Castle package that will work in Medium Trust?  I'm attempting to do everything with Express versions, but they don't allow full-blown VS Solutions, so a bunch of the References break within the solutions.  As such, I can't recompile NHibernate with the new Castle dlls.</p><p>I see a lot of potential with NHibernate and I've invested about two weeks trying to learn NHibernate and get it to work in Medium Trust.  I even jumped into MyGeneration and learned to use the tool to generate POCOs and XML mapping files (wasn't happy with the existing templates as they didn't seem to convert data types well...a lot of Unknown types were present in the outputfiles).  But I am about ready to give up and go back to SubSonic.</p><p>Eventually I will probably get on board and buy VS Professional but w/ VS 2010 just around the corner it doesn't make sense to make the investment now.  In addition, I'll be releasing my project as Open Source and expect the majority of users to be using Express Editions and shared hosting.</p><p>The whole Medium Trust thing is frustrating though.  I can understand hosts needing to protect their servers and their clients' instances, but even MS fails to test their code to make sure it runs in Medium Trust.  An example is the Entity Framework...currently you need to compile your edmx file in a seperate strongly typed class to get it to work. Annoying to say the least.</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 15 2009, 12:16</em>

---

<p>@samk: you can get the latest compiled binaries from SVN: <a href="https://cuyahoga.svn.sourceforge.net/svnroot/cuyahoga/trunk/lib" rel="nofollow noopener" title="https://cuyahoga.svn.sourceforge.net/svnroot/cuyahoga/trunk/lib">https://cuyahoga.svn.sourceforge.net/svnroot/cuyahoga/trunk/lib</a></p>

**Comment by disqus-comments-migrator** [comment]
<em>Fabio Maulo commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 16 2009, 05:11</em>

---

<p>did you try NHPG ?<br><a href="http://nhforge.org/wikis/proxygenerators10/faq.aspx" rel="nofollow noopener" title="http://nhforge.org/wikis/proxygenerators10/faq.aspx">http://nhforge.org/wikis/proxygenerators10/faq.aspx</a></p>

**Reply by disqus-comments-migrator** [reply]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 16 2009, 12:39, in reply to Fabio Maulo's [comment](https://github.com/martijnboland/blog/discussions/32#discussioncomment-8952827)</em>

---

<p>NHPG is a workaround for the issue that is now fixed. We shouldn't be needing this anymore.</p>

**Comment by disqus-comments-migrator** [comment]
<em>Bill Pierce commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 16 2009, 09:34</em>

---

<p>One line code change in DP to support Medium Trust?  Man I feel stupid now.</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 16 2009, 10:04</em>

---

<p>Not exactly one line. A few months ago there has been another change in DP, but also (and much more important): since .NET 3.5 (or .NET 2.0 SP1) the required permissions for Reflection.Emit have been changed.</p>
